tests/test_biascorr.py

```diff
@@ -0,0 +1,518 @@
+        assert np.nanvar(corrected_dem - self.ref) < 0.01 * np.nanvar(synthetic_bias)
+
+    def test_deramp(self) -> None:
+        """Test the subclass Deramp."""
+
+        # Try default "fit" parameters instantiation
+        deramp = biascorr.Deramp()
+
+        assert deramp._fit_or_bin == "fit"
+        assert deramp._meta["fit_func"] == polynomial_2d
+        assert deramp._meta["fit_optimizer"] == scipy.optimize.curve_fit
+        assert deramp._meta["poly_order"] == 2
+
+    @pytest.mark.parametrize("order", [1, 2, 3, 4])  # type: ignore
+    def test_deramp__synthetic(self, order: int) -> None:
+        """Run the deramp for varying polynomial orders using a synthetic elevation difference."""
+
+        # Get coordinates
+        xx, yy = np.meshgrid(np.arange(0, self.ref.shape[1]), np.arange(0, self.ref.shape[0]))
+
+        # Number of parameters for a 2D order N polynomial called through np.polyval2d
+        nb_params = int((order + 1) * (order + 1))
+
+        # Get a random number of parameters
+        np.random.seed(42)
+        params = np.random.normal(size=nb_params)
+
+        # Create a synthetic bias and add to the DEM
+        synthetic_bias = polynomial_2d((xx, yy), *params)
+        bias_dem = self.ref - synthetic_bias
+
+        # Fit
+        deramp = biascorr.Deramp(poly_order=order)
+        deramp.fit(reference_dem=self.ref, dem_to_be_aligned=bias_dem, subsample=10000, random_state=42)
+
+        # Check high-order parameters are the same within 10%
+        fit_params = deramp._meta["fit_params"]
+        assert np.shape(fit_params) == np.shape(params)
+        assert np.allclose(
+            params.reshape(order + 1, order + 1)[-1:, -1:], fit_params.reshape(order + 1, order + 1)[-1:, -1:], rtol=0.1
+        )
+
+        # Run apply and check that 99% of the variance was corrected
+        corrected_dem = deramp.apply(bias_dem)
+        assert np.nanvar(corrected_dem - self.ref) < 0.01 * np.nanvar(synthetic_bias)
+
+    def test_terrainbias(self) -> None:
+        """Test the subclass TerrainBias."""
+
+        # Try default "fit" parameters instantiation
+        tb = biascorr.TerrainBias()
+
+        assert tb._fit_or_bin == "bin"
+        assert tb._meta["bin_sizes"] == 100
+        assert tb._meta["bin_statistic"] == np.nanmedian
+        assert tb._meta["terrain_attribute"] == "maximum_curvature"
+
+    def test_terrainbias__synthetic(self) -> None:
+        """Test the subclass TerrainBias."""
+
+        # Get maximum curvature
+        maxc = xdem.terrain.get_terrain_attribute(self.ref, attribute="maximum_curvature")
+
+        # Create a bias depending on bins
+        synthetic_bias = np.zeros(np.shape(self.ref.data))
+
+        bin_edges = np.array((-1, 0, 0.1, 0.5, 2, 5))
+        bias_per_bin = np.array((-5, 10, -2, 25, 5))
+        for i in range(len(bin_edges) - 1):
+            synthetic_bias[np.logical_and(maxc.data >= bin_edges[i], maxc.data < bin_edges[i + 1])] = bias_per_bin[i]
```

Comment:
Could you comment on what this does? My soupy brain can't follow

Reply:
Done!